### DISCUSSION TITLE
No more "Issues"

### DISCUSSION THREAD

**Discussion by antonmedv** [deployphp/deployer#3888] [General]
Hey Deployer folks!

I’ve decided to drop the concept of “Issues” from this repo because I believe it’s flawed, and GitHub’s “Discussions” feature handles things better.

Here are my two main problems with issues:

1. People use them as a catch-all for bugs, features, and questions, which leads to extra work sorting through them.
2. The “open” or “closed” status can cause frustration. People sometimes feel slighted when an issue is closed, interpreting it as a negative response from maintainers.

Switching to dedicated “Features” and “Bugs” groups in Discussions will improve things. With no “open issues” counter on the repo’s main page, there’s less pressure to close things too soon.

Plus, both I and the community can prioritize issues better now because they can be upvoted—a great feature missing from Issues (which led to those “+1” comments).

Lastly, Discussions will focus non-contribution activity in one place, meaning more eyes on questions and ideas.

**Comment by hashimaziz1** [comment]
Weren't you on another repo just a few weeks ago complaining about this behaviour for an issue you created, and now you're doing it yourself on your own repo? You even copied that dev's announcement post word for word, do they know about that? I disagreed in that case and I disagree now - issues have emojis so they don't need upvotes, and the only "pressure" is self-imposed - the open issue count is *motivation* to fix bugs, as I've seen they have helped you fix many bugs in the software and improve Deployer massively over time. I can't see any benefit to this move at all.

**Reply by antonmedv** [reply]
> Weren't you on another repo just a few weeks ago complaining about this behaviour for an issue you created

No. 

> You even copied that dev's announcement post word for word, do they know about that?

Yes, copied. In my own words. The repo I copied this idea was livewire. And to copy idea is okay. Livewire devs, I copied your idea. 

> I disagreed in that case and I disagree now

It is okay. You have a right to be disagree. 

> I've seen they have helped you fix many bugs in the software and improve Deployer massively over time.

Issues are the way of keeping track of bugs/features/discussions/questions. Lately the become unmanageable and duplicated by discussions. 

>  I can't see any benefit to this move at all.

We will see how a new approach works and decide later on

**Comment by pniederlag** [comment]
I can understand your motivation and reasoning. There is always a decision wether someone elses problem is solved, irrelevant or worth fixing/implementing which is a bunch of work and responsibility put on the maintainer.

Also I have already seen a bunch of knowledge beeing hidden in closed bugs und bug-reports that are just mere support-requests

So, happy to see you take a different approach, looking forward to see how it turns out in the long run.

**Reply by pniederlag** [reply]
Maybe it's a good idea to either drop the issues tab completly or to add a single issue pointing to this discussion... currently I see 0 open issues on the issues tab.

**Reply by antonmedv** [reply]
I thought about hiding "Issues" tab completely. But this can lead to confusion on there to open an issue. Now people can still do to "Issues" tab, and click "New issue" button to be redirected to "Discussions". I think this is a good compromise for now.

**Comment by alexisLefebvre** [comment]
What about the issues that were "open"?

Like this one:

- ~~https://github.com/deployphp/deployer/issues/3118~~
  - https://github.com/deployphp/deployer/discussions/3893 

Do you plan to migrate the issues to discussions so that people can upvote them?

**Reply by antonmedv** [reply]
Yes. I added a tag (still open) to all remaining issues and have plans to go through all of them. Fix issues or convert to a discussion if needed.

**Reply by hashimaziz1** [reply]
Not true, you closed [this one](https://github.com/deployphp/deployer/issues/3371) automatically and it's still an ongoing issue.

**Reply by antonmedv** [reply]
I've added label `still open` to the mentioned issue. I hope this resolves the issue.

**Comment by rabol** [comment]
If it's issues or discussion, is not a big deal to me as long as things get's fixed, if it's a bug, or there is a chance for new features

I have had open questions regarding permissions for many month and no responses /solution. I have even offered @antonmedv to pay him to help me with one issue, but that did not work

Have a look at the open PR's some is from 2023, so people are willing to help, but as PR's are not merged people looses interest in helping.

The request about Ubuntu 22.04 has been there for more than one year and still no ETA

I for one would love to help, but I have not the needed skill to do it, so for now I'sponsoring' and let's see how that helps the project and me

**Reply by antonmedv** [reply]
Thanks for reaching out and for your support. I understand your frustration with the delays, and unfortunately, there’s no ETA on some of the requests like the one for Ubuntu 22.04, as this is an open-source project with contributions happening as time permits. However, if you’d like to discuss paid help for specific issues, we can certainly chat about it and see how I can assist you.
I appreciate your understanding and willingness to support the project—every bit helps!
